### PR TITLE
[Telink] Add Bridge example app demo

### PR DESCRIPTION
**Pull request by interfer**
**Problem**
- These changes add new Telink app example

**Change overview**
- Added Bridge App example (4 Lights and a Temperature Sensor)

**Testing**
Tested manually with chip-tool.

**Steps**:

- Run: $ chip-tool pairing ble-thread <...>
- Wait till success

### CODE REVIEW

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

**Review by arkq — APPROVED** [review]

**Comment by s07641069** [comment]
@bzbarsky-apple Could you review/approve PR?

**Review by bzbarsky-apple — COMMENTED** [review]
But now are the spec-required ClusterRevision attributes even present?  I don't see them in the cluster declarations, and I don't see values being produced for them...

**Comment by interfer** [comment]
@bzbarsky-apple good point, thanks, added `ClusterRevision` attribute to clusters

File: examples/bridge-app/telink/src/AppTask.cpp [hunk 1]
```diff
@@ -0,0 +1,1094 @@
...
+    else if ((attributeId == FeatureMap::Id) && (maxReadLength == 4))
+    {
+        uint32_t featureMap;
+        FeatureMap::Get(dev->GetEndpointId(), &featureMap);
+        memcpy(buffer, &featureMap, sizeof(featureMap));
+    }
+    else if ((attributeId == ClusterRevision::Id) && (maxReadLength == 2))
+    {
+        uint16_t clusterRevision;
+        ClusterRevision::Get(dev->GetEndpointId(), &clusterRevision);
```
Comment (bzbarsky-apple) [inline examples/bridge-app/telink/src/AppTask.cpp:713]: This is just going to call right back into this function, no?  Was this tested?

File: examples/bridge-app/telink/src/AppTask.cpp [hunk 2]
```diff
@@ -0,0 +1,1094 @@
...
+DECLARE_DYNAMIC_ATTRIBUTE_LIST_BEGIN(descriptorAttrs)
+DECLARE_DYNAMIC_ATTRIBUTE(Clusters::Descriptor::Attributes::DeviceTypeList::Id, ARRAY, kDescriptorAttributeArraySize,
+                          0), /* device list */
+    DECLARE_DYNAMIC_ATTRIBUTE(Clusters::Descriptor::Attributes::ServerList::Id, ARRAY, kDescriptorAttributeArraySize,
+                              0), /* server list */
+    DECLARE_DYNAMIC_ATTRIBUTE(Clusters::Descriptor::Attributes::ClientList::Id, ARRAY, kDescriptorAttributeArraySize,
+                              0), /* client list */
+    DECLARE_DYNAMIC_ATTRIBUTE(Clusters::Descriptor::Attributes::PartsList::Id, ARRAY, kDescriptorAttributeArraySize,
+                              0), /* parts list */
+    DECLARE_DYNAMIC_ATTRIBUTE(Clusters::Descriptor::Attributes::ClusterRevision::Id, INT16U, ZCL_DESCRIPTOR_CLUSTER_REVISION,
```
[the remark this answers is not in the document]
Comment (interfer) [reply examples/bridge-app/telink/src/AppTask.cpp:212]: Just right, added conditions to handle `ClusterRevision` and FeatureMap in other clusters.
But it's not clear for me what did You mean with 
> So it's possible that DECLARE_DYNAMIC_ATTRIBUTE_LIST_END includes this

File: examples/bridge-app/telink/src/AppTask.cpp (hunk 1 above)
Comment (interfer) [reply examples/bridge-app/telink/src/AppTask.cpp]: It's been falling in infinite loop indeed, fixed it already, thanks. Tested manually with chip-tool, all 3 clusters have now correct respond on `read cluster-revision` command 
> ./chip-tool onoff read cluster-revision {NODE_ID} {ENDPOINT_ID}

> 
>  			AttributeDataIB =
>  			{
>  				DataVersion = 0x6c62e585,
> 				AttributePathIB =
> 				{
>  					Endpoint = 0x6,
>  					Cluster = 0x6,
>  					Attribute = 0x0000_FFFD,
>  				}
>  					
>  				Data = 4, 
> 			},

File: examples/bridge-app/telink/src/AppTask.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply examples/bridge-app/telink/src/AppTask.cpp:212]:
```
#define DECLARE_DYNAMIC_ATTRIBUTE_LIST_END()                                                                                       \
    {                                                                                                                              \
        ZAP_EMPTY_DEFAULT(), 0xFFFD, 2, ZAP_TYPE(INT16U), ZAP_ATTRIBUTE_MASK(EXTERNAL_STORAGE)                                     \
...
```
0xFFFD is the attribute id of ClusterRevision, right?

File: examples/bridge-app/telink/src/AppTask.cpp [hunk 3]
```diff
@@ -357,6 +358,14 @@ EmberAfStatus HandleReadBridgedDeviceBasicAttribute(Device * dev, chip::Attribut
         MutableByteSpan zclNameSpan(buffer, maxReadLength);
         MakeZclCharString(zclNameSpan, dev->GetName());
     }
+    else if ((attributeId == FeatureMap::Id) && (maxReadLength == 4))
+    {
+        *buffer = (uint32_t) ZCL_BRIDGED_DEVICE_BASIC_INFORMATION_FEATURE_MAP;
```
Comment (bzbarsky-apple) [inline examples/bridge-app/telink/src/AppTask.cpp:363]: This is wrong, exactly in the way it was wrong in the initial PR; see https://github.com/project-chip/connectedhomeip/pull/25798#discussion_r1146679558

Please look at the actual types of the things involved here and think about what this does.

What really confuses me is that your code for the temperature sensor is right, while this should-be-the-same code is wrong....

File: examples/bridge-app/telink/src/AppTask.cpp [hunk 4]
```diff
@@ -357,6 +358,14 @@ EmberAfStatus HandleReadBridgedDeviceBasicAttribute(Device * dev, chip::Attribut
...
+    }
+    else if ((attributeId == ClusterRevision::Id) && (maxReadLength == 4))
+    {
+        *buffer = (uint16_t) ZCL_BRIDGED_DEVICE_BASIC_INFORMATION_CLUSTER_REVISION;
```
Comment (bzbarsky-apple) [inline examples/bridge-app/telink/src/AppTask.cpp:367]: This is broken too.

File: examples/bridge-app/telink/src/AppTask.cpp [hunk 5]
```diff
@@ -373,6 +382,10 @@ EmberAfStatus HandleReadOnOffAttribute(Device * dev, chip::AttributeId attribute
     {
         *buffer = dev->IsOn() ? 1 : 0;
     }
+    else if ((attributeId == Clusters::OnOff::Attributes::ClusterRevision::Id) && (maxReadLength == 4))
+    {
+        *buffer = (uint16_t) ZCL_ON_OFF_CLUSTER_REVISION;
```
Comment (bzbarsky-apple) [inline examples/bridge-app/telink/src/AppTask.cpp:387]: And this.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/bridge-app/telink/src/AppTask.cpp (hunk 3 above)
Comment (interfer) [reply examples/bridge-app/telink/src/AppTask.cpp]: Ok, I hope this time I got it right. Modified it in latest commit

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**